### DISCUSSION TITLE
NER model fails to detect if first word is an entity unless a non-whitespace character is added to the start of the text

### DISCUSSION THREAD

**Discussion by AdirthaBorgohain** [explosion/spaCy#9785] [Help: Model Advice] [answered]
So I trained a NER model using spacy with the help of transformers model and I am facing this weird issue.  Suppose my input text is: 
Endothelial cells (HAECs) with nicotine resulted in NLRP3 ASC inflammasome activation.

In this case, the model can detect all entities as expected (eg. nicotine as drug, NLRP3 as gene). But it fails to detect Endothelial cells as anatomy. However if I add any non-whitespace character like a comma at the start of the text (for eg: ,Endothelial cells (HAECs) with nicotine resulted in NLRP3 ASC inflammasome activation.), then it correctly detects Endothelial cells as anatomy this time. What can be the reason for this? I have been looking around but I still am not able to figure out a possible reason for this.

**Comment by adrianeboyd** [comment, marked as answer]
That sounds weird! I wonder whether it could be something related how to the tokens are aligned during processing.

Does your model predict correctly if you train and evaluate on a small number of examples that include this text as one of the examples? Just as a sanity check, see what happens if you train and evaluate on the same ~10 examples, in order to rule out that it's a systematic alignment problem of some sort.

Actually, I remember that we've seen some similar-looking problems with `PERSON` entities as single names like "John" instead of "John Smith", and I can see similar behavior with `en_core_web_trf` with "John" at the beginning of a sentence, although even this is weird and depends on both the particular name and the following context. I suppose it could be something related to the NER model state for the first token in the text, although this clearly isn't a systematic bug, either, since I don't see it with other non-`trf` `en_core_web_*` models. It is a bit baffling...

**Reply by AdirthaBorgohain** [reply]
Hi there. Sorry for the late response. I kind of figured out the issue. The issue was with my parsing script to build the .spacy files for training. I had an unnecessary assertion line to check ``spans.get('start')`` under an if statement. So, wherever the start index was zero, it was getting ignored completely and wasn't getting into the .spacy file. Hence the issue. Thanks for the response.
